workspace: Persist window values without project

mchisolm0 opened the pull request:
Persist and restore window values (size, position, etc.) to the KV Store when there are no projects open.

Relates to Discussion https://github.com/zed-industries/zed/discussions/24228#discussioncomment-15224666

Release Notes:

- Added persistence for window size when no projects are open

mchisolm0 committed:
workspace: Persist window values without project

Persist and restore window values (size, position, etc.) to the KV Store
when there are no projects open.

ConradIrwin commented on crates/gpui/src/platform.rs:
```diff
@@ -1258,7 +1258,7 @@ pub(crate) struct WindowParams {
 
 /// Represents the status of how a window should be opened.
-#[derive(Debug, Copy, Clone, PartialEq)]
+#[derive(Debug, Copy, Clone, PartialEq, Deserialize, Serialize)]
```
The disadvantage of adding Serialize/Deserialize here is that we can never change this structure without breaking serialization. Although it's a bit of extra code (though hopefully nothing beyond what AI can build) it's less likely that we'll break things in the future if we have a specific structure/enum in the serialization code that gets converted to this (maybe we can copy the existing stuff from workspace? though tbh storing a JSON blob seems better than destructuring to a bunch of columns).

Not a huge issue though, so we could leave this and then deal with it when we want to change this enum.

mchisolm0 committed:
workspace: Use no-project window fallback for remote workspace

Switch remote workspace without a project to fallback to window bounds
stored in the KV store like local workspaces now do.
